sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -221,7 +221,8 @@ case class FileSourceScanExec(
           val sortColumns =
             spec.sortColumnNames.map(x => toAttribute(x)).takeWhile(x => x.isDefined).map(_.get)
 
-          val sortOrder = if (sortColumns.nonEmpty) {
+          val sortOrder = if (sortColumns.nonEmpty &&
+            !partitionFilters.exists(ExecSubqueryExpression.hasSubquery)) {
```

Comment:
why is it needed?

Reply:
Because outputPartitioning and outputOrdering are used before execution (`EnsureRequirements`) and `partitionFilters` can't be evaluated before execution if they have subquery expressions, otherwise we get a `Subquery ... has not finished` exception.

Reply:
`partitionFilters.exists(ExecSubqueryExpression.hasSubquery)` is used more than once, maybe we can create
```
private def partitionOnlyAvailableAtRuntime: Boolean = {
  partitionFilters.exists(ExecSubqueryExpression.hasSubquery)
}
```

Reply:
Ok, done.